lockc/src/maps.rs

```diff
@@ -0,0 +1,144 @@
+        let ap = accessed_path::new(allowed_path_s)?;
+        allowed_paths_mount_baseline.insert(i as u32, ap, 0)?;
+    }
+
+    let mut allowed_paths_access_restricted: HashMap<_, u32, accessed_path> =
+        bpf.map_mut("allowed_paths_access_restricted")?.try_into()?;
+    for (i, allowed_path_s) in SETTINGS.allowed_paths_access_restricted.iter().enumerate() {
+        let ap = accessed_path::new(allowed_path_s)?;
+        allowed_paths_access_restricted.insert(i as u32, ap, 0)?;
+    }
+
+    let mut allowed_paths_access_baseline: HashMap<_, u32, accessed_path> =
+        bpf.map_mut("allowed_paths_access_baseline")?.try_into()?;
+    for (i, allowed_path_s) in SETTINGS.allowed_paths_access_baseline.iter().enumerate() {
+        let ap = accessed_path::new(allowed_path_s)?;
+        allowed_paths_access_baseline.insert(i as u32, ap, 0)?;
+    }
+
+    Ok(())
+}
+
+pub fn add_container(
+    bpf: &mut Bpf,
+    container_id: String,
+    pid: i32,
+    policy_level: container_policy_level,
+) -> Result<(), MapOperationError> {
+    debug!(
+        "adding container {} with pid {} policy level {} to eBPF map",
+        container_id, pid, policy_level as u32
+    );
+
+    let mut containers: HashMap<_, container_id, container> =
+        bpf.map_mut("containers")?.try_into()?;
+    let container_key = container_id::new(&container_id)?;
+    let container = container { policy_level };
+    containers.insert(container_key, container, 0)?;
+
+    let mut processes: HashMap<_, i32, process> = bpf.map_mut("processes")?.try_into()?;
+    let process = process {
+        container_id: container_key,
+    };
+    processes.insert(pid, process, 0)?;
+
+    Ok(())
+}
+
+pub fn delete_container(bpf: &mut Bpf, container_id: String) -> Result<(), MapOperationError> {
+    debug!("deleting container {} from eBPF map", container_id);
+
+    let mut containers: HashMap<_, container_id, container> =
+        bpf.map_mut("containers")?.try_into()?;
+    let container_key = container_id::new(&container_id)?;
+    containers.remove(&container_key)?;
+
+    let processes: HashMap<_, i32, process> = bpf.map("processes")?.try_into()?;
+    let mut processes_mut: HashMap<_, i32, process> = bpf.map_mut("process")?.try_into()?;
+    for res in processes.iter() {
+        let (pid, process) = res?;
+        if process.container_id.id == container_key.id {
+            processes_mut.remove(&pid)?;
+        }
+    }
+
+    Ok(())
+}
+
+pub fn add_process(bpf: &mut Bpf, container_id: String, pid: i32) -> Result<(), MapOperationError> {
+    debug!(
+        "adding process {} (container: {}) to eBPF map",
+        pid, container_id
+    );
+
+    let mut processes: HashMap<_, i32, process> = bpf.map_mut("processes")?.try_into()?;
+    let container_key = container_id::new(&container_id)?;
+    let process = process {
+        container_id: container_key,
+    };
+    processes.insert(pid, process, 0)?;
+
+    Ok(())
```

Comment:
nitpicking: not needed, use the result from the previous line

Reply:
Same here, I would need to explicitly convert the error from previous line with `map_err`. I prefer staying with `Ok(())`